src/main/java/com/limechain/runtime/hostapi/HostApi.java

```diff
@@ -94,4 +94,25 @@ protected static ByteBuffer getByteBuffer(Memory memory) {
         }
         return buffer != null ? buffer : memory.buffer();
     }
+
+    public byte[] getDataFromMemory(RuntimePointerSize runtimePointerSize) {
+        Memory memory = getMemory();
+        byte[] data = new byte[runtimePointerSize.size()];
+        memory.buffer().get(runtimePointerSize.pointer(), data, 0, runtimePointerSize.size());
+        return data;
+    }
+
+    public RuntimePointerSize addDataToMemory(byte[] data) {
+        Memory memory = getMemory();
+        ByteBuffer buffer = getByteBuffer(memory);
+        int position = buffer.position();
+        buffer.put(position, data, 0, data.length);
+        buffer.position(position + data.length);
+
+        return new RuntimePointerSize(position, data.length);
+    }
+
+    public void putDataToMemoryBuffer(RuntimePointerSize runtimePointerSize, byte[] data) {
+        //TODO: implement
```

Comment:
This is not in the scope of this PR?

Reply:
Yes, a separate PR will cover memory operations.